untraced/ykpack/src/types.rs

```diff
@@ -136,6 +124,30 @@ impl Ty {
     }
 }
 
+#[cfg(target_pointer_width = "64")]
+impl Ty {
+    // Pack size & align into 64 bits (48 bits for size, 16 bits for align).
```

Comment:
I agree that this is good for performance, but it'd be irresponsible of me to not ask: how future proof is the 48-bit address space model?

Reply:
It's not an address: it's a size. If you manage to make a static type which exceeds 48 bits (~131072GiB) in size, I will be very impressed. More to the point, you'll get a compile-time error as ykpack will refuse to encode it, so there isn't a correctness issue (though there is a usability issue).

Reply:
As long as we are not going to become the next "no-one will need more than 640KB of RAM" :)

It should be easily changed later I hope anyway.

Reply:
I think we're a little way off from machines that can have 48 bits worth of RAM in, and we'll probably have some advanced notice of that fact :) Fortunately we can change the representation transparently: that's the beauty of using methods not attributes :)